src/App.js

```diff
@@ -1,23 +1,27 @@
-import logo from './logo.svg';
-import './App.css';
+import { Routes, Route } from 'react-router-dom';
+import { useDispatch } from 'react-redux';
+import { useEffect } from 'react';
+import Header from './components/Header';
+import Missions from './components/Missions';
+import Rockets from './components/Rockets';
+import Profile from './components/Profile';
+import { getMission } from './redux/missions/mission';
+import { getData } from './redux/rockets/rockets';
 
 function App() {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(getMission);
```

Comment:
According to the project requirements:

![image](https://user-images.githubusercontent.com/57421823/156241853-e43f032c-3e8b-4e16-9003-0a28ee6bf619.png)

Therefore:

- [ ]  Please, make sure the missions' data is fetched only when the user navigates to the missions page.

## Explanation:

This is to make sure the API data is only consumed as needed.

## Hints:

You could remove this `getMission()` function call from this `App.js`, and add instead another `useEffect` to the main missions component, so on the first render of the missions, it fetches them, and not before.  

Please, remember to dispatch the data to the Redux store when the missions are fetched.

That way, we meet the requirements and we'll move closer to approval 💪